src/adc.rs

```diff
@@ -1,18 +1,75 @@
-///! busy-waiting because the transfers should complete at approximately the same time.
+///! # Design
+///!
+///! Stabilizer ADCs are connected to the MCU via a simplex, SPI-compatible interface. The ADCs
+///! require a setup conversion time after asserting the CSn (convert) signal to generate the ADC
+///! code from the sampled level. Once the setup time has elapsed, the ADC data is clocked out of
+///! MISO. The internal setup time is managed by the SPI peripheral via a CSn setup time parameter
+///! during SPI configuration, which allows offloading the management of the setup time to hardware.
+///!
+///! Because of the SPI-compatibility of the ADCs, a single SPI peripheral + DMA is used to automate
+///! the collection of multiple ADC samples without requiring processing by the CPU, which reduces
+///! overhead and provides the CPU with more time for processing-intensive tasks, like DSP.
+///!
+///! The automation of sample collection utilizes two DMA streams, the SPI peripheral, and a timer
+///! compare channel for each ADC. The timer comparison channel is configured to generate a
+///! comparison event every time the timer is equal to a specific value. Each comparison then
+///! generates a DMA transfer event to write into the SPI TX buffer. Although the SPI is a simplex,
+///! RX-only interface, it is configured in full-duplex mode and the TX pin is left disconnected.
+///! This allows the SPI interface to periodically read a single word whenever a word is written to
+///! the TX side. Thus, by running a continuous DMA transfer to periodically write a value into the
+///! TX FIFO, we can schedule the regular collection of ADC samples in the SPI RX buffer.
+///!
+///! In order to collect the acquired ADC samples into a RAM buffer, a second DMA transfer is
+///! configured to read from the SPI RX FIFO into RAM. The request for this transfer is connected to
+///! the SPI RX data signal, so the SPI peripheral will request to move data into RAM whenever it is
+///! available. When enough samples have been collected, a transfer-complete interrupt is generated
+///! and the ADC samples are available for processing.
+///!
+///! The SPI peripheral internally has an 8- or 16-byte TX and RX FIFO, which corresponds to a 4- or
+///! 8-sample buffer for incoming ADC samples. During the handling of the DMA transfer completion,
+///! there is a small window where buffers are swapped over where it's possible that a sample could
+///! be lost. In order to avoid this, the SPI RX FIFO is effectively used as a "sample overflow"
+///! region and can buffer a number of samples until the next DMA transfer is configured. If a DMA
+///! transfer is still not set in time, the SPI peripheral will generate an input-overrun interrupt.
+///! This interrupt then serves as a means of detecting if samples have been lost, which will occur
+///! whenever data processing takes longer than the collection period.
+///!
+///!
```

Comment:
Correct - it does not increase us to 4 + batch_size because DBM means that there's always a DMA transfer configured with the SPI peripheral. Since a DMA transfer is always draining the SPI, the internal SPI FIFO can never fill beyond a single sample.

Thus, if we have a batch size of 1-3, we would actually have less time to service the DMA transfer completion in DBM